File: go-sdk/sdk/path_utils/path_utils.go
package path_utils

import (
	"github.com/go-logr/logr"
	"github.com/spf13/viper"
	path2 "path"
)

type PathUtils struct {
	logger logr.Logger
}

func NewPathUtils(logger logr.Logger) *PathUtils {
	return &PathUtils{
		logger: logger.WithName("[PATH UTILS]"),
	}
}

func (pu PathUtils) GetBasePath() string {
	return viper.GetString("metadata.base_path")
}
func (pu PathUtils) ComposePath(relativePath ...string) string {
	path := pu.GetBasePath()
	if len(relativePath) == 0 {
		return path
	}

	for _, elem := range relativePath {
		path = path2.Join(path, elem)
	}
	return path
}

File: py-sdk/src/sdk/path_utils/path_utils.py
import os
from dataclasses import dataclass

from loguru import logger
from loguru._logger import Logger
from vyper import v


@dataclass
class PathUtils:
    logger: Logger = logger.bind(context="[PATH UTILS]")

    @staticmethod
    def get_base_path() -> str:
        return v.get("metadata.base_path")

    @staticmethod
    def compose_path(*relative_path)-> str:
        base_path = PathUtils.get_base_path()
        if not relative_path:
            return base_path

        return os.path.join(base_path, *relative_path)
